elytron/src/main/java/org/wildfly/extension/elytron/FileSystemRealmDefinition.java

```diff
@@ -157,13 +198,29 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
 
             final String hashEncoding = HASH_ENCODING.resolveModelAttribute(context, model).asString();
             final String hashCharset = HASH_CHARSET.resolveModelAttribute(context, model).asString();
+            final String credentialStore = CREDENTIAL_STORE.resolveModelAttribute(context, model).asStringOrNull();
+            final String secretKey = SECRET_KEY.resolveModelAttribute(context, model).asStringOrNull();
+
+
+            String credentialStoreRuntimeCapability;
+            ServiceName credentialStoreServiceName = null;
+            if(credentialStore != null) {
+                credentialStoreRuntimeCapability = buildDynamicCapabilityName(CREDENTIAL_STORE_CAPABILITY, credentialStore);
+                credentialStoreServiceName = context.getCapabilityServiceName(credentialStoreRuntimeCapability, CredentialStore.class);
+            }
+
+            SecretKey key = null;
+            if(credentialStore != null && secretKey != null) {
+                key = getKey(context, credentialStore, secretKey);
+            }
+            SecretKey finalKey = key;
 
             final InjectedValue<PathManager> pathManagerInjector = new InjectedValue<>();
             final InjectedValue<NameRewriter> nameRewriterInjector = new InjectedValue<>();
 
+
             TrivialService<SecurityRealm> fileSystemRealmService = new TrivialService<>(
                     new TrivialService.ValueSupplier<SecurityRealm>() {
-
```

Comment:
It would be good to remove these white space changes.

Reply:
Removed